modules/templates/templates_api.rb

```diff
@@ -14,9 +14,9 @@ class Proxy::TemplatesApi < Sinatra::Base
     end
   end
 
-  get "/:kind" do |kind|
+  get "/:kind/?:template?/?:hostgroup?" do |kind, template, hostgroup|
```

Comment:
I think creating a dedicated route for hostgroup templates would be a better approach: the ```template``` and ```hostgroup``` parameters will no longer be optional, which would allow for better handling of malformed urls (atm it's  valid to pass ```kind``` and ```template``` parameters only).